DWARF uses poor sizeof style

so on my unsupported macOS platform, in the middle of `cd trans; bmake -r` I see a warning from `dw2_basic.c`

```
==> Compiling src/common/dwarf2/dw2_basic.c
cc -D TRANS_X86 -I /Users/drj/prj/tendra/trans/src/x86 -I /Users/drj/prj/tendra/trans/src/common -D TDF_DIAG4 -D DWARF2 -D STABS -I .. -I .  -I /Users/drj/prj/tendra/trans/include  -c dw2_basic.c -o /Users/drj/prj/tendra/trans/obj/src/x86/_partial/src/common/dwarf2/dw2_basic.o
dw2_basic.c:76:18: warning: shift count >= width of type [-Wshift-count-overflow]
                        value |= - (1 << ((sizeof(value) * 8) - 7));
                                      ^  ~~~~~~~~~~~~~~~~~~~~~~~~~
dw2_basic.c:100:18: warning: shift count >= width of type [-Wshift-count-overflow]
                        value |= - (1 << ((sizeof(value) * 8) - 7));
                                      ^  ~~~~~~~~~~~~~~~~~~~~~~~~~
2 warnings generated.
```

Now, as I know and as i am pleased to see reinforced in `/STYLE`, `sizeof` is not a function.

So this should read

    value |= - (1 << ((sizeof value * 8) - 7));

or, if parens are needed to reinforce the precedence order:

    value |= - (1 << (((sizeof value) * 8) - 7));

Personally, I think I _very slightly_ prefer the latter.

That change however, would not address the warning. _That_ problem is (presumably) that literal `1` does not have the same type as _value_, `long`.

--- a/trans/src/common/dwarf2/dw2_basic.c
+++ b/trans/src/common/dwarf2/dw2_basic.c
@@ -73,7 +73,7 @@ sleb128(long value)
 		if (negative) {
 			/* sign extend, since C doesn't imply arithmetic
 			 * shift */
-			value |= - (1 << ((sizeof(value) * 8) - 7));
+			value |= - (1L << (((sizeof value) * 8) - 7));
 		}
 		/* sign bit of byte is 2nd high order bit (0x40) */
 		if (value == - (long)((byt & 0x40) != 0)) {
@@ -97,7 +97,7 @@ sleb128_length(long value)
 		if (negative) {
 			/* sign extend, since C doesn't imply arithmetic
 			 * shift */
-			value |= - (1 << ((sizeof(value) * 8) - 7));
+			value |= - (1L << (((sizeof value) * 8) - 7));
 		}
 		if (value == - (long)((byt & 0x40) != 0)) {
 			return op;

